docs/testing-overview.md

```diff
@@ -2,11 +2,26 @@
 
 Gutenberg contains both PHP and JavaScript code, and encourages testing and code style linting for both.
 
+## Why test?
+
+Aside from the joy testing will bring to your life, tests are important not only because they help to ensure that our application behaves as it should, but also because they provide concise examples of how to use a piece of code. 
+
+Tests are also part of our code base, which means we apply to them the same standards we apply to all our application code. 
+
+As with all code, tests have to be maintained. Writing tests for the sake of having a test isn't the goal – rather we should try to strike the right balance between covering expected and unexpected behaviours, speedy execution and code maintenance.
```

Comment:
I like this sentence, a lot of people seem to be "dogmatic" about testing everything no matter what :)

Reply:
I like it a lot, too. Props to @ramonjd 💯